code/_onclick/item_attack.dm

```diff
@@ -31,8 +31,16 @@
 	var/messagesource = M
 
 	if (can_operate(M))  //Checks if mob is lying down on table for surgery
-		if (do_surgery(M,user,src))
-			return 0
+		if(istype(src,/obj/item/robot_parts))//popup ovveride for direct attach
+			if(!attempt_initiate_surgery(src, M, user,1))
+				return 0
+		if(istype(src,/obj/item/weapon/screwdriver) && M.get_species() == "Machine")
+			if(!attempt_initiate_surgery(src, M, user))
+				return 0
+		if(is_sharp(src))
+			if(!attempt_initiate_surgery(src, M, user))
+				return 0
+		return
```

Comment:
@Aurorablade remove the return? add 
``` javascript
else
    return 1
```
to all of the if(!attempt_initiate_surgery) checks above?